concordia/views.py

```diff
@@ -209,25 +211,54 @@ def registration_rate(self, group, request):
 @method_decorator(never_cache, name="dispatch")
 class ConcordiaRegistrationView(RatelimitMixin, RegistrationView):
     form_class = UserRegistrationForm
+    ratelimit_group = "registration"
     ratelimit_key = "ip"
     ratelimit_rate = registration_rate
     ratelimit_method = "POST"
-    ratelimit_block = True
+    ratelimit_block = settings.RATELIMIT_BLOCK
 
 
 @method_decorator(never_cache, name="dispatch")
 class ConcordiaLoginView(RatelimitMixin, LoginView):
+    ratelimit_group = "login"
     ratelimit_key = "ip"
     ratelimit_rate = "3/15m"
     ratelimit_method = "POST"
-    ratelimit_block = True
+    ratelimit_block = False
+
+    def post(self, request, *args, **kwargs):
+        """
+        Handle POST requests: instantiate a form instance with the passed
+        POST variables and then check if it's valid.
+        """
+        form = self.get_form()
+        if form.is_valid():
```

Comment:
I'm wondering how much we should care about the ability to guess using parallel requests here since the success path isn't limited. I'm wondering whether this process should always test the rate-limit but instead use a CAPTCHA for clients which are rate-limited:

```python
blocked = is_ratelimited()
recent_captcha = time.time() - request.session.get("captcha_validation_time", 0) > 86400

if blocked:
    if not recent_captcha:
        # redirect to a captcha or add a captcha field to the form so the next attempt will have the session value
    else:
        raise RateLimited
else:
    return super().post(request, *args, **kwargs)
```

Reply:
I'd like to address this in a separate ticket.

Reply:
#654 created for this

Reply:
We can defer this but then I'm wondering whether we should either leave `RATELIMIT_BLOCK = True` or increase the limits because this change is effectively removing the reason for rate-limiting this view since an attacker guessing passwords still learns when they have the correct password.

Reply:
I see what you mean. I guess I should take care of this today then

Reply:
Okay, so digging into this a bit more, I don't follow the logic. If the login request is rate limited (blocked) and the user has completed a captcha, it shouldn't raise a RateLimited exception. In fact if we allow people to complete a captcha for rate limited logins then the RateLimited exception would never come up, would it? So I think it should be more like the following. Also, the test for recent_captcha should be that the difference is less than 86400, not greater than

```
blocked = is_ratelimited()
recent_captcha = time.time() - request.session.get("captcha_validation_time", 0) < 86400

if blocked:
    if not recent_captcha:
        # redirect to a captcha or add a captcha field to the form so the next attempt will have the session value
else:
    return super().post(request, *args, **kwargs)
```

Reply:
Hmmm, yes, that needed more time. The idea was that we never actually test the password before checking the CAPTCHA to avoid leaking that information but your simplified version does that.